src/util/TriState.hpp

```diff
@@ -33,6 +33,7 @@
 #include <cstdint>
 
 #ifdef _WIN32
+#include <winsock2.h>
```

Comment:
sincos???

Reply:
Ok, I have to split the commit! 
This is because you have to inlude winsock2.h before windows.h.

Reply:
> This is because you have to inlude winsock2.h before windows.h.

Is that a documented WIN32 API requirement?